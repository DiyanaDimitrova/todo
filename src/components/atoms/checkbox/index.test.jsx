import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Checkbox from './index';

describe('Checkbox Component', () => {
  const mockOnChange = jest.fn();

  test('should match snapshot', () => {
    const { asFragment } = render(
      <Checkbox label="Test" checked={true} onChange={mockOnChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should call onChange when clicked', () => {
    render(<Checkbox label="Test" checked={true} onChange={mockOnChange} />);
    
    fireEvent.click(screen.getByLabelText(/test/i)); // Use label text for better accessibility
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });

  test('should not call onChange if not provided', () => {
    const { container } = render(<Checkbox label="Test" checked={true} />);
    
    // eslint-disable-next-line testing-library/no-node-access
    fireEvent.click(container.firstChild);
    // There should be no error, but no call to onChange
    expect(mockOnChange).not.toHaveBeenCalled();
  });
});
