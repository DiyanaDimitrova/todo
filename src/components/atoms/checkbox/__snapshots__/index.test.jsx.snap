// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkBoxWrapper"
  >
    <label
      class="label"
      for="Test"
    >
      <input
        checked=""
        class="input"
        id="Test"
        type="checkbox"
      />
      <span>
        Test
      </span>
    </label>
  </div>
</DocumentFragment>
`;
